modules/auxiliary/scanner/ssl/ssl_version.rb

```diff
@@ -126,19 +174,20 @@ def print_cert(cert, ip)
       print_status("\tNot Valid After: #{cert.not_after}")
 
       # Checks for common properties of self signed certificates
+      # regex tried against a bunch of alexa top 100 and others.
+      # https://rubular.com/r/Yj6vyy1VqGWCL8
       caissuer = nil
       cert.extensions.each do |e|
-        e = e.to_s
-        if /CA Issuers - URI:([^, \n]*)/i.match(e)
-          caissuer = /CA Issuers - URI:([^, \n]*)/i.match(e)
-          break
-        end
+        next unless /CA Issuers - URI:([^, \n]*)/i =~ e.to_s
+
+        caissuer = ::Regexp.last_match(1)
+        break
       end
 
       if caissuer.nil?
```

Comment:
This won't change anything:

```
irb(main):001:1> if /CA Issuers - URI:([^, \n]*)/i ~= 'a'
irb(main):002:1>   print "B"
irb(main):003:0> end
irb(main):007:0> ::Regexp.last_match(1)
=> ""
irb(main):008:0> 
irb(main):009:0> ::Regexp.last_match(1).nil?
=> false
irb(main):010:0> 
```

We should likely be using `.blank?` to check things here to account for NIL as well as blank strings.

Reply:
Resolved in https://github.com/rapid7/metasploit-framework/pull/17149/commits/24a467b0cfd69f52bd503081844f376ea15c9e7b